src/core/Xtarget.ml

```diff
@@ -13,4 +13,14 @@ type t = {
   lazy_content : string lazy_t;
   (* This is valid only for xlang = Xlang.L ..., not for LRegex|LGeneric *)
   lazy_ast_and_errors : (AST_generic.program * Tok.location list) lazy_t;
+  lockfile_data : lockfile_data option;
+}
+
+and lockfile_data = {
```

Comment:
Maybe we should use a variant then? or not abuse Xtarget.ml and have a separate type?
src/core/Lockfile.ml ?